### PR TITLE
Fix the Sentry error for `eth_getBalance`

### LINKED ISSUE #10551: Error: JsonRpcEngine: Response has no error or result for request:
Sentry Issue: [METAMASK-GP1K](https://sentry.io/organizations/metamask/issues/2078124553/?referrer=github_integration)

```
Error: JsonRpcEngine: Response has no error or result for request:
{
  "id": 6902237914659141,
  "jsonrpc": "2.0",
  "params": [ ...  ],
  "method": "eth_getBalance"
}
  at "stack": "Error: JsonRpcEngine: Response has no error or result for request:\n{\n  \"id\": 6902237914659141,\n  \"jsonrpc\": \"2.0\",\n  \"params\": [ ...  ],\n  \"method\": \"eth_getB... (metamask/background.js:60:376981)\n    at Function._checkForCompletion ()
```

### PR DESCRIPTION
**Pull request by filipsekulic**
## Explanation
Fix the Sentry error for `eth_getBalance`.

## More Information
Fixes https://github.com/MetaMask/metamask-extension/issues/10551

### Before

https://user-images.githubusercontent.com/92310504/173789548-7a09a2a9-562e-45af-b17e-debf9ed386d3.mov

### After

https://user-images.githubusercontent.com/92310504/173789772-d8e8157d-f7e2-485a-8685-37138a715488.mov

## Manual Testing Steps

1. Import a network (i.e. Fantom Opera). You can use [Chainlist](https://chainlist.org/).
2. Connect to that network.
3. Check how many accounts you have (`n` accounts).
4. Open background console.
5. Disconnect the Internet (from the Network tab).
6. Lock MM.
7. Click `Forgot Password` and wait for the error to be shown (disconnected).
8. Import your SRP and create a new password.
9. Create `n+1` accounts.
10. There is no `eth_getBalance` error in the background console.

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

**Comment by darkwing** [comment]
@brad-decker Do we want to move this logic to https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L2113-L2130 ?  Or is it best handled like this PR, only in specific places ?

**Comment by brad-decker** [comment]
Placing the logic where you specified would not mitigate the failure here because we are calling into the provider, versus using the method on the metamask-controller. We probably want to bake some offline-support, or network-failure fallback/graceful failure into the provider stack *at some point* but for mitigating this particular issue I think the proposed solution is a good patch.

File: app/scripts/lib/account-tracker.js [hunk 1]
```diff
@@ -243,8 +243,19 @@ export default class AccountTracker {
    * @returns {Promise} after the account balance is updated
    */
   async _updateAccount(address) {
+    let balance;
```
Comment (darkwing) [inline app/scripts/lib/account-tracker.js:246]:
```suggestion
    let balance = '0x0';
```

**Review by darkwing — CHANGES_REQUESTED** [review]
Made a small suggestion -- works great though!

**Comment by filipsekulic** [comment]
Hey @darkwing! 
Thanks! Updated. 🙂

File: app/scripts/lib/account-tracker.js [hunk 2]
```diff
@@ -243,8 +243,19 @@ export default class AccountTracker {
...
+
     // query balance
-    const balance = await this._query.getBalance(address);
+    try {
+      balance = await this._query.getBalance(address);
+    } catch (error) {
+      if (error.data.request.method === 'eth_getBalance') {
```
[the remark this answers is not in the document]
Comment (filipsekulic) [reply app/scripts/lib/account-tracker.js]: Fixed here: aaad1377a1bba0b96a47ab7a17f69541af851960

**Review by darkwing — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]
LGTM

**Pull request merged**